utils/parenthesesutils.go

```diff
@@ -55,12 +55,26 @@ func RemovePlaceholderParentheses(pattern, target string) string {
 	return temp
 }
 
-// Escaping Parentheses with no corresponding placeholder.
-func addEscapingParentheses(pattern, target string) string {
+// AddEscapingParenthesesForUploadCmd escaping parentheses with no corresponding placeholder.
+func AddEscapingParenthesesForUploadCmd(pattern, target, targetPathInArchive string) string {
```

Comment:
The phrase "cmd" isn't the correct one. This library does not include commands.
I therefore suggest renaming this function to `addEscapingParenthesesForUpload` and move it into the `upload.go` files which is the only one that uses it.